EIPS/eip-7701.md

```diff
@@ -118,15 +137,27 @@ Execution starts at the first byte of code section 0, and pc is set to 0
 ```
 
 However, if a contract is referenced in an `AA_TX_TYPE` transaction as a `Sender`, `Paymaster` or a `Deployer`,
-execution starts at the first byte of code section with the `entry_point_role` marker corresponding to the current step,
+execution starts at the first byte of code section with the current `entry_point_role` variable value,
 and `pc` is set to `0`.
 
-If the specified contract does not contain such a section, or is not an EOF contract, the transaction is not valid.
+If the specified contract does not contain such a section, or is not an EOF contract,
+execution starts at code section 0, and pc is set to 0.
+
+The transaction is considered invalid if it never reached a code section corresponding to the current `entry_point_role`
+in any contract called through an uninterrupted chain of `DELEGATECALL`/`EXTDELEGATECALL` calls.
+
+The transaction is considered invalid if it reached such a code section but did not
+perform [explicit acceptance](#accepting-the-transaction-by-sender-or-paymaster).
 
 The `target_section_flags` parameter is added to provide signaling into the EVM so that EOF can perform some additional
 validations as part of EOF code validation.
 The description of specific flags and their impact on EOF validation shall be added as a separate EIP.
 
+### Accepting the transaction by `Sender` or `Paymaster`
+
+In order for the `Sender` or `Paymaster` contracts to accept the transaction,
+they must perform a `CALL` to the `AA_ORIGIN` address with the appropriate acceptance data as call data.
```

Comment:
Does the call have to be performed by the actual `Sender` / `Paymaster`, or just in its context?  If it's the former then the above proxy-support method, then proxies won't work.  If the latter, then the security concerns I wrote above apply.

Reply:
I think supporting existing proxy contracts is essential, so the call must be made in the context of the `Sender` / `Paymaster`. As I replied, I don't think there is a security concern unless there is a `DELEGATECALL` to an untrusted contract, in which case the attack does not require messing with roles.